### PR TITLE
chore: #144 - sunsetting modal component 

### PR DESCRIPTION
**Pull request by ss-dimasm**
# Pull request checklist

**Context**

Refactor the Modal (Dialog) component work will be done gradually due to need to keep the objective review, here what I've plan for this series while moving forward:

- **Add a `Dialog` component and style alignment**
This involves creating a new component based on the `HTMLDialogElement` interface, we will implement common `dialog` functionalities here. The component's initial styling will be adjusted to ensure it aligns correctly with the design

- **Tweak the design and did a code clean up**
In this part, the design of the `Dialog` component will be more refined based on the question that raised to the Design Team and their specification. Including with the using or declaring the CSS variable for `Dialog`, 
Additionally, the code will be cleaned up to improve readability and maintainability based on the reviewer suggestions

- **Final Documentation for the `Dialog` component**
The `Dialog` component will be documented to provide detailed information about its usage and functionality. Finalise the checklist that has been made in the ticket is included

- **Sunsetting the `Modal` component**
The existing `Modal` component will be deprecated

---

This PR was the fourth part of this series, that will focus to `Modal` component deprecation. We're skipping the third part due to incomplete of design documentation for Dialog component. [See the reason](https://github.com/reapit/elements/issues/144#issuecomment-2434380522)

**Detail as per issue below (required):**
- Mark some styles and interfaces from `Modal` as deprecated
- Left a `console.warn` method inside of `Modal`  component  to indicate the deprecation for the next major release (perhaps v6? 🤔)
- Write a migration guidance from `Modal` to `Dialog` component

### CODE REVIEW

File: src/components/modal/modal.mdx [hunk 1]
```diff
@@ -5,10 +5,26 @@ import * as ModalStories from './modal.stories'
...
+
+- The `Dialog` component offers a more comprehensive set of features, including customization options, and accessibility enhancements
+- Using the `Dialog` component aligns with our commitment to providing a consistent and modern user experience across our product line
+
+**Migration Guidance:**
+
+- Identify `Modal` Usage: Thoroughly review your codebase to locate all instances of the `Modal` component
+- Evaluate `Dialog` Suitability: Assess whether the `Dialog` component's features and capabilities meet your specific requirements
+- Transition Gradually: If feasible, gradually replace `Modal` components with `Dialog` components in a phased approach to minimize disruption
+- Leverage `Dialog` Features: Explore the `Dialog` component's extensive documentation and examples to effectively utilize its functionalities, such as custom titles, content, actions, and full-screen mode
```
Comment (kurtdoherty) [inline src/components/modal/modal.mdx:26]: **question:** I'm not aware of any "full-screen mode" for Dialog? Where's that coming from?

File: src/components/modal/modal.atoms.tsx
Comment (kurtdoherty) [inline src/components/modal/modal.atoms.tsx:1]: **todo:** The styled components like `ElModalBg` should all be deprecated too please 🙏  Same with the `useModal` hook

File: src/components/modal/modal.mdx
Comment (kurtdoherty) [inline src/components/modal/modal.mdx:1]: **thought:** When I think of migration docs, I generally think of before and after style code examples and content that highlights the key differences between the old thing and the new thing. The content here sounds more like marketing content with phrases like "the more versatile and feature-rich Dialog", "extensive documentation"; it feels a bit "fluffy" to me and I wonder if we could just be more direct 🤔

File: src/components/modal/modal.tsx [hunk 2]
```diff
@@ -34,6 +40,8 @@ export const Modal: FC<ModalProps> = ({ isOpen, onModalClose, title, className,
 
   if (!isOpen) return null
 
+  console.warn('Modal component will be removed in future major version. Use `Dialog` component instead')
```
Comment (kurtdoherty) [inline src/components/modal/modal.tsx:43]: **todo:** Logging content to the browser's console is a side-effect of rendering, so it should be inside a `useEffect`.

**Review by kurtdoherty — CHANGES_REQUESTED** [review]
Thanks @ss-dimasm! Just a few minor requests.

File: src/components/modal/modal.mdx (hunk 1 above)
Comment (ss-dimasm) [reply src/components/modal/modal.mdx:26]: >  I'm not aware of any "full-screen mode" for Dialog? Where's that coming from?

Sorry to make the confusion, I mean that's the backdrop pseudo from HTMLDialogElement, it's act like a "full-screen"

File: src/components/modal/modal.atoms.tsx
Comment (ss-dimasm) [reply src/components/modal/modal.atoms.tsx:1]: i missed that, thanks

File: src/components/modal/modal.mdx
Comment (ss-dimasm) [reply src/components/modal/modal.mdx:1]: I just realized, we've missed the important part for the "migration" documentation, which it must have the before and after code example. I appreciate your opinion

File: src/components/modal/modal.mdx (hunk 1 above)
Comment (kurtdoherty) [reply src/components/modal/modal.mdx:26]: Ah, right. I'd avoid the phrase "full-screen mode" here, as that's not what the backdrop represents.

**Review by ss-dimasm — COMMENTED** [review]
hi @kurtdoherty, thanks for the heads up!

I have revised the PR to more closely align with your suggestions

**Review by kurtdoherty — APPROVED** [review]

**Pull request merged**